mitosheet/src/mito/Mito.tsx

```diff
@@ -279,15 +279,21 @@ export const Mito = (props: MitoProps): JSX.Element => {
         }
     }, [mitoAPI, sendFunctionStatus])
 
+    // We're storing the last analysisData in a ref so that we can check the
+    // analysisData's code against the code in the cell and make sure we aren't
+    // overwriting any changes the user might have made. 
+    const oldCodeRef = useRef(analysisData.code);
     useEffect(() => {
         /**
          * We only write code after the render count has been incremented once, which
          * means that we have read in and replayed the updated analysis, etc. 
          */
         if (analysisData.renderCount >= 1) {
             // Finially, we can go and write the code!
-            props.jupyterUtils?.writeGeneratedCodeToCell(analysisData.analysisName, analysisData.code, userProfile.telemetryEnabled, analysisData.publicInterfaceVersion);
+            props.jupyterUtils?.writeGeneratedCodeToCell(analysisData.analysisName, analysisData.code, userProfile.telemetryEnabled, analysisData.publicInterfaceVersion, oldCodeRef?.current);
         }
+        // After using the ref to get the old code, we update it to the newest analysis.
+        oldCodeRef.current = analysisData.code;
```

Comment:
Should this be up, inside the if statement? If no, why not.

Reply:
Oh yeah I think it's better to just update it any time the analysis data changes.